zingolib/src/blaze/block_management_reorg_detection.rs

```diff
@@ -446,17 +446,22 @@ impl BlockManagementData {
         self.wait_for_block(after_height).await;
 
         {
-            // Read Lock
+            // Read Lock for
             let blocks = self.blocks_in_current_batch.read().await;
+            // take the height of the first/highest block in the batch and subtract target height to get index into the block list
             let pos = blocks.first().unwrap().height - after_height;
             match nf {
                 PoolNullifier::Sapling(nf) => {
                     let nf = nf.to_vec();
 
+                    // starting with a block and searching until the top of the batch
```

Comment:
I think the code is much more readable when not broken up by many comments. mabye describe the block in one comment above?